zeppelin-web/src/app/interpreter/interpreter.controller.js

```diff
@@ -22,6 +22,171 @@ angular.module('zeppelinWebApp').controller('InterpreterCtrl',
     $scope.showRepositoryInfo = false;
     $scope._ = _;
 
+    // auto complete related
+    $scope.suggestions = [];
+    $scope.selectIndex = -1;
+    var selectedUser = '';
+    var selectedUserIndex = 0;
+    var previousSelectedList = [];
+    var searchText = [];
+
+    function updatePreviousList() {
+      for (var i = 0; i < searchText.length; i++) {
+        previousSelectedList[i] = searchText[i];
+      }
+    }
+
+    function convertToString(setting) {
```

Comment:
the name `convertToString` implies the following signature `Array -> String`, at least in my opinion. This implementation, however as a signature `Object -> Void` which makes it hard for reader to understand what's happening. a more appropriate name would be something like `assignToUsers`; it should also at least receive stringified searchText as an argument